modules/nf-core/orthofinder/main.nf

```diff
@@ -18,22 +19,28 @@ process ORTHOFINDER {
     task.ext.when == null || task.ext.when
 
     script:
-    def args    = task.ext.args ?: ''
-    prefix      = task.ext.prefix ?: "${meta.id}"
+    def args   = task.ext.args   ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    def include_command = prior_run   ? "-b $prior_run" : ''
+
     """
     mkdir temp_pickle
 
     orthofinder \\
-        $args \\
         -t $task.cpus \\
         -a $task.cpus \\
         -p temp_pickle \\
         -f input \\
-        -n $prefix
+        -n $prefix \\
+        $include_command $args
+
+    if [ -e input/OrthoFinder/Results_$prefix ]; then
+        mv input/OrthoFinder/Results_$prefix $prefix
+    fi
 
-    mv \\
-        input/OrthoFinder/Results_$prefix \\
-        $prefix
+    if [ -e WorkingDirectory/OrthoFinder/Results_$prefix ]; then
```

Comment:
Thank you for these enhancements @chriswyatt1

Does `WorkingDirectory` here mean that to correctly use this module in the resume mode, the data from the previous run has to be placed in a folder named `WorkingDirectory`? Can we make the name dynamic by using the  `prior_run` variable here?

Reply:
Sorry, I was away these last few days. 
Yes, you are right:

WorkingDirectory is the automatic name that Orthofinder creates inside the results folder. 
So when you run without a previous "WorkingDirectory", currently the script writes to the folder  (-f input), then a folder called "OrthoFinder/" and then to a test folder  (-n $prefix). Complicated,... so it then moves this back to the current directory:
`mv input/OrthoFinder/Results_$prefix $prefix`

In the `Results_$prefix` folder, you find the `WorkingDirectory`.

When you run with the resume flag, you need to supply the WorkingDirectory  in (-b)`input/OrthoFinder/Results_$prefix/WorkingDirectory`, which is moved to `$prefix/WorkingDirectory`, in that mv statement above. This then saves the output to mv `WorkingDirectory/OrthoFinder/Results_$prefix` (when running resume). So it runs out of the previously run folder. 

In (long) summary, we can use: `if [ -e ${prior_run}/OrthoFinder/Results_$prefix ]; then`

Reply:
Thank you for the detailed response.